### PR TITLE
Feature/collectd addon

### PR DESCRIPTION
**Pull request by BrianHicks**
- [x] Installs cleanly on a fresh build of most recent master branch
- [ ] Upgrades cleanly from the most recent release
- [x] Updates documentation relevant to the changes
- [ ] Rebases cleanly onto the latest master

Testing in progress, I just wanted to get this out here so we can have a discussion: this PR removes logstash's collectd forwarding behavior. Motivation is on 05c49f1. Does anyone have feedback of a better way to do this?

### CODE REVIEW

File: addons/collectd.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+- hosts: localhost
```
Comment (siddharthist) [inline addons/collectd.yml]: Should this really be `localhost`?

**Comment by siddharthist** [comment]
Unfortunately, this fails with our Zookeeper checks (which check for `/etc/collectd.d/zookeeper.conf`).

**Comment by siddharthist** [comment]
Looks good otherwise, the addon playbook installs correctly and all services remain healthy. We'll just have to remove that distributive check, which will be part of the distributive refactor that's on its way.

**Comment by BrianHicks** [comment]
@siddharthist is there a way we could remove that check now? Or move it to the collectd checks right away?

**Comment by siddharthist** [comment]
It is installed via the monolithic distributive-rpm. If you merge my PRs on mantl-packaging, I can put the new distributive PR in right away, we can merge that and then this. 

Or, we could remove `distributive.yml` from the zookeeper role altogether in the meantime.

**Comment by BrianHicks** [comment]
Rebased onto latest master

**Comment by siddharthist** [comment]
This will be unblocked by #1268, which doesn't check for `/etc/collectd.d/zookeeper.conf`.

**Comment by ryane** [comment]
now that #1268 is in, we can probably get this merged as soon as the merge conflicts are resolved

**Comment by BrianHicks** [comment]
Great, I’ll work on that today!

**Comment by BrianHicks** [comment]
this passes most checks but fails on DO for some reason, for different failure scenarios. Weird.

File: roles/collectd/tasks/mesos-agent.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+---
+- name: check if mesos agent is present
+  stat:
+    path: /etc/sysconfig/mesos-agent
+  register: mesos_agent_present
```
Comment (ryane) [inline roles/collectd/tasks/mesos-agent.yml]: is this being used anywhere?

File: roles/collectd/tasks/mesos-master.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+---
+- name: check if mesos master is present
+  stat:
+    path: /etc/sysconfig/mesos-master
+  register: mesos_master_present
```
Comment (ryane) [inline roles/collectd/tasks/mesos-master.yml]: same here?

**Comment by siddharthist** [comment]
rebased

**Comment by siddharthist** [comment]
@BrianHicks We've seen the most recent DO failure in CI builds before, is that what you were referring to? Is this ready to merge after we fix those variables that @ryane pointed out?

**Comment by ryane** [comment]
@BrianHicks are the `mesos_agent_present` `mesos_master_present` variables being used anywhere?

**Comment by BrianHicks** [comment]
```shell
$ ag mesos_agent_present
roles/collectd/tasks/mesos-agent.yml
5:  register: mesos_agent_present

$ ag mesos_master_present
roles/collectd/tasks/mesos-master.yml
5:  register: mesos_master_present
```

so... no? Do you reckon they ought to be removed?

**Comment by ryane** [comment]
I kinda think so. they don't break anything but just so there is no confusion...

**Comment by siddharthist** [comment]
I changed that, squashed the commit, and rebased. If tests pass, I'll merge.

**Comment by siddharthist** [comment]
Well, they all failed, but not for related reasons.

**Comment by ryane** [comment]
tested successfully on gce

**Pull request merged**